code/jobs/SESQueuedMail.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/**
+ * SESQueuedMail
+ *
+ * @author Stephen McMahon <stephen@silverstripe.com.au>
+ */
+class SESQueuedMail extends AbstractQueuedJob implements QueuedJob {
+
+	public function __construct($destinations, $subject, $rawMessageText) {
+		$this->To = $destinations;
+		$this->Subject = $subject;
+		$this->RawMessageText = $rawMessageText;
+	}
+
+	public function getTitle() {
+
+		return 'Email To: ' . implode(', ', $this->To) . ' Subject: ' . $this->Subject;
+	}
+
+	public function getJobType() {
+
+		$this->totalSteps = '1';
+
+		return QueuedJob::QUEUED;
+	}
+
+	/**
+	 * Lets process a single node, and publish it if necessary
+	 */
+	public function process() {
+
+		if ($this->isComplete) {
+
+			return;
+		}
+
+		// we need to always increment! This is important, because if we don't then our container
+		// that executes around us thinks that the job has died, and will stop it running.
+		$this->currentStep++;
+
+		try {
+			$response = Injector::inst()->get('SESMailer')->sendSESClient($this->To, $this->RawMessageText);
+		} catch (\Aws\Ses\Exception\SesException $ex) {
+			$this->addMessage($ex, 'ERR');
+			$this->JobStatus = QueuedJob::STATUS_BROKEN;
+
+			return;
+		}
+
+		if (isset($response['MessageId']) && strlen($response['MessageId'])) {
```

Comment:
Shouldn't isComplete be set _regardless_ of whether $response[messageid] is set? If not, it'll simply retry the send again; this isn't necessarily a bad thing, but you should probably also use the totalSteps and currentStep values so it never does a retry more than X number of times (3 is probably a reasonable default)